nitransforms/base.py

```diff
@@ -150,18 +160,25 @@ def resample(self, moving, order=3, mode='constant', cval=0.0, prefilter=True,
             slightly blurred if *order > 1*, unless the input is prefiltered,
             i.e. it is the result of calling the spline filter on the original
             input.
+
         Returns
         -------
         moved_image : `spatialimage`
             The moving imaged after resampling to reference space.
+
         """
         moving_data = np.asanyarray(moving.dataobj)
         if output_dtype is None:
             output_dtype = moving_data.dtype
 
+        moving_grid = ImageGrid(moving)
+
+        def _map_indexes(ijk):
+            return moving_grid.inverse.dot(self.map(self.reference.affine.dot(ijk)))
```

Comment:
I think we can reduce this down to defining this concatenation, where the affines of the moving and reference grids are applied on input indexes.